src/server/scripts/Events/midsummer.cpp

```diff
@@ -172,6 +504,25 @@ struct npc_midsummer_torch_target : public ScriptedAI
 // SPELLS
 ///////////////////////////////
 
+class spell_bonfires_blessing : public AuraScript
+{
+    PrepareAuraScript(spell_bonfires_blessing)
+
+    void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (!IsHolidayActive(HOLIDAY_FIRE_FESTIVAL))
+        {
+            if (Unit* target = GetTarget())
+                target->RemoveAurasDueToSpell(SPELL_BONFIRES_BLESSING);
```

Comment:
You need to validate spells used within spell scripts. See Bool Validate(SpellInfo* spellInfo)

Reply:
Alright, I was unsure if it's necessary if the spell is just being used for removeAura()

Reply:
always, becuase if it for some reason wouldnt exist the server would crash